src/main/java/org/prebid/server/floors/BasicPriceFloorResolver.java

```diff
@@ -518,6 +544,25 @@ private static Price roundPrice(Price price) {
         return price != null ? Price.of(price.getCurrency(), BidderUtil.roundFloor(price.getValue())) : null;
     }
 
+    private static Price resolvePrice(Price effectiveFloor, Price convertedFloorMin, Price effectiveFloorMin) {
+        final BigDecimal floorValue = effectiveFloor.getValue();
+        final String floorCurrency = effectiveFloor.getCurrency();
+
+        final BigDecimal floorMinValue = convertedFloorMin.getValue();
+        final String floorMinCurrency = convertedFloorMin.getCurrency();
+
+        if (StringUtils.equals(floorCurrency, floorMinCurrency) && floorValue != null && floorMinValue != null) {
+            if (floorValue.compareTo(floorMinValue) > 0) {
+                return roundPrice(effectiveFloor);
+            } else {
+                return roundPrice(convertedFloorMin);
+            }
```

Comment:
Can be simplified to `return floorValue.compareTo(floorMinValue) > 0 ? roundPrice(effectiveFloor) : roundPrice(convertedFloorMin)` 

We can even insert ternary into the single call to `roundPrice` but i think it's unreadable.